src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/VisualStudioMacLSPEditorFeatureDetector.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.Editor.Razor;
+using MonoDevelop.Core.FeatureConfiguration;
+using MonoDevelop.Ide;
+using MonoDevelop.Projects;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Shared]
+    [Export(typeof(LSPEditorFeatureDetector))]
+    internal class VisualStudioMacLSPEditorFeatureDetector : LSPEditorFeatureDetector
+    {
+        private const string RazorLSPEditorFeatureFlag = "Razor.LSP.Editor";
+        private const string DotNetCoreCSharpCapability = "CSharp&CPS";
```

Comment:
Nit: This is not a "DotNetCore" capability, just an SDK-style project one.

But also, VS Mac doesn't use CPS. I assume this still works because it fakes the capability?

Reply:
[Yup!](https://github.com/xamarin/vsmac/blob/064fd35f26a025fcf5c233abbc1e876bb86afad9/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs#L1369-L1372)